### PR TITLE
ui_rgw: Fix the returned port in rgw gatways URL endpoints

### PR DESCRIPTION
**Pull request by rjfd**
Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

File: srv/modules/runners/ui_rgw.py [hunk 1]
```diff
@@ -97,8 +97,16 @@ def _urls(self):
                 self.credentials['urls'].append(cached[minion]['rgw_endpoint'])
                 return
 
+        port = '7480'  # civetweb default port
+        if os.path.exists("/srv/salt/ceph/configuration/files/ceph.conf.rgw"):
+            with open("/srv/salt/ceph/configuration/files/ceph.conf.rgw") as rgw_conf_file:
+                for line in rgw_conf_file:
+                    if line:
+                        match = re.search(r'rgw.*frontends.*=.*"civetweb.*port=(\d+)"', line)
```
Comment (theanalyst) [inline srv/modules/runners/ui_rgw.py]: doesn't look like a valid regex?

File: srv/modules/runners/ui_rgw.py [hunk 2]
```diff
@@ -97,8 +97,16 @@ def _urls(self):
                 self.credentials['urls'].append(cached[minion]['rgw_endpoint'])
                 return
 
+        port = '7480'  # civetweb default port
+        if os.path.exists("/srv/salt/ceph/configuration/files/ceph.conf.rgw"):
+            with open("/srv/salt/ceph/configuration/files/ceph.conf.rgw") as rgw_conf_file:
+                for line in rgw_conf_file:
+                    if line:
```
Comment (theanalyst) [inline srv/modules/runners/ui_rgw.py]: I would assume for line in file would already not return None, if you're checking for emptiness (probably not necessary as the re.search would not return?)

File: srv/modules/runners/ui_rgw.py [hunk 3]
```diff
@@ -97,15 +97,23 @@ def _urls(self):
...
+        for rgw_conf_file_path in glob.glob("/srv/salt/ceph/configuration/files/ceph.conf.*"):
+            if os.path.exists(rgw_conf_file_path) and os.path.isfile(rgw_conf_file_path):
+                with open(rgw_conf_file_path) as rgw_conf_file:
+                    for line in rgw_conf_file:
+                        if line:
+                            match = re.search(r'rgw.*frontends.*=.*port=(\d+)(s?)', line)
+                            if match:
+                                port = match.group(1)
+                                ssl = match.group(2)
+                                found = True
```
Comment (theanalyst) [inline srv/modules/runners/ui_rgw.py]: can't we break here itself?

**Comment by rjfd** [comment]
@theanalyst sorry, just update the PR with some changes to the implementation

File: srv/modules/runners/ui_rgw.py (hunk 1 above)
Comment (rjfd) [inline srv/modules/runners/ui_rgw.py]: I tested and it works good

File: srv/modules/runners/ui_rgw.py (hunk 2 above)
Comment (rjfd) [inline srv/modules/runners/ui_rgw.py]: probably right. We will clean it up afterwards

File: srv/modules/runners/ui_rgw.py [hunk 4]
```diff
@@ -97,15 +97,23 @@ def _urls(self):
...
 
+        port = '7480'  # civetweb default port
+        ssl = ''
+        found = False
+        for rgw_conf_file_path in glob.glob("/srv/salt/ceph/configuration/files/ceph.conf.*"):
+            if os.path.exists(rgw_conf_file_path) and os.path.isfile(rgw_conf_file_path):
+                with open(rgw_conf_file_path) as rgw_conf_file:
+                    for line in rgw_conf_file:
+                        if line:
+                            match = re.search(r'rgw.*frontends.*=.*port=(\d+)(s?)', line)
```
Comment (theanalyst) [inline srv/modules/runners/ui_rgw.py]: not sure whether this is a valid python regex?

**Pull request merged**